pkg/build/util/generate_test.go

```diff
@@ -335,20 +412,39 @@ func TestSubstituteImageSTIMatch(t *testing.T) {
 	if bc.Parameters.Strategy.STIStrategy.Image != originalImage {
 		t.Errorf("STI BuildConfig was updated when Build was modified")
 	}
+}
 
+func TestSubstituteImageRepositorySTIMatch(t *testing.T) {
+	source := mockSource()
+	strategy := mockSTIStrategyForImageRepository()
+	repoRef := strategy.STIStrategy.From
+	output := mockOutput()
+	bc := mockBuildConfig(source, strategy, output)
+	build := GenerateBuildFromConfig(bc, nil, nil, nil)
+
+	// STI build with a matched base image
+	// base image should be replaced
+	SubstituteImageRepoReferences(build, repoRef, newImage)
+	if build.Parameters.Strategy.STIStrategy.Image != newImage {
+		t.Errorf("Base image name was not substituted in sti strategy")
+	}
+	if bc.Parameters.Strategy.STIStrategy.From.Name != imageRepoName ||
+		bc.Parameters.Strategy.STIStrategy.From.Namespace != imageRepoNamespace {
```

Comment:
this should be checking that the bc.Parameters.Strategy.STIStrategy.Image field is not changed.

the original issue was that when we assigned the build.Parameters.STrategy.STIStrategy.Image field, we accidentally also changed the buildconfig field.  You're not updating the From field anywhere so there's no reason to think it would be incorrectly modified (though leaving the check is fine)